### ISSUE TITLE
Hide imports from users

### ISSUE THREAD

**Issue by ksunden** [wright-group/WrightTools#173]
Currently we do all of our imports in the simplest fashion: 
```
import os
import copy
etc
```

This leads to these imports being visible to the end users, which is less than ideal

It appears that the standard way of getting around this is to use:
```
import os as _os
import copy as _copy
```

This will make things like tab completion and the like ignore the imports

This change is tedious, but not difficult to undertake

**Comment by untzag** [comment]
I found [this](https://stackoverflow.com/questions/7423744/how-should-i-perform-imports-in-a-python-module-without-polluting-its-namespace) stackoverflow thread thought provoking.

For inspiration I looked to the scipy module to see how they handled namespace pollution. The answer is actually pretty interesting.

Scipy maintains a bunch of smaller modules, each with library imports etc. Each of these modules defines `__all__`. Then, the `__init__` file in each subpackage does an import * on each module.

https://github.com/scipy/scipy/blob/v0.18.1/scipy/interpolate/__init__.py

I'm not sure what the consequences of this strategy would be, but it does seem pretty elegant. I like that it forces a design with many minimal modules, rather than a few large ones. Of course, it would take a day or so to change over---and who knows what it would do to auto-generated.
